Server/src/main/java/org/openas2/partner/XMLPartnershipFactory.java

```diff
@@ -248,19 +258,58 @@ public void loadPartnership(Map<String, Object> partners, List<Partnership> part
     }
 
     /**
-     * Appends the passed element as a child of the root in the partnership document.
+     * Appends the passed element as a child of the root in the partnership
+     * document.
      * It does NOT check if the passed element is a valid element.
+     * 
      * @param newElement - the element to be added.
      */
     public void addElement(Element newElement) {
         Document doc = getPartnershipsXml();
+        logger.debug("addElement **********" + doc.toString());
+        logElement(newElement);
         Node importedNode = doc.importNode(newElement, true);
         doc.getDocumentElement().appendChild(importedNode);
+        try {
+            storePartnership();
+        } catch (OpenAS2Exception e) {
+            logger.error("Failed to store partnership: " + e.getMessage());
+            // Handle the exception as needed
+        }
+
+    }
+
+    private void logElement(Element element) {
+        logger.info("@logElement");
+        logger.info("Element: " + element.getTagName());
+
+        // Log attributes of the current element
+        NamedNodeMap attributes = element.getAttributes();
+        for (int i = 0; i < attributes.getLength(); i++) {
+            Node attribute = attributes.item(i);
+            logger.info("Attribute: " + attribute.getNodeName() + "=" + attribute.getNodeValue());
```

Comment:
All INFO, DEBUG and TRACE level logging that has string formatting within the log message argument must be wrapped in a check to see if the logging is on or off because otherwies it evaluates the string before calling the log statement which is a significant waste of time of that level of logging is off.